src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int START_CAPACITY = 10;
+    private int size;
+    private Object[] elements;
+
+    public ArrayList() {
+        size = 0;
+        elements = new Object[START_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (elements.length == size) {
+            growCapacity();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexValidation(index);
+        if (elements.length == size) {
+            growCapacity();
+        }
+        if (elements.length > index) {
```

Comment:
we already check index on line 30
we don't need this

Reply:
fixed